File: haru-engine/src/main/java/haru/engine/port/adapter/web/TaskController.java
package haru.engine.port.adapter.web;

import lombok.Value;
import org.springframework.web.bind.annotation.GetMapping;
import org.springframework.web.bind.annotation.RestController;

import java.util.Arrays;
import java.util.List;

/**
 * @author takeshi
 */
@RestController
public class TaskController {

  @Value
  public static class TaskJson {
    private final String name;
  }

  @GetMapping("/api/v1/tasks")
  public List<TaskJson> getTasks() {
    return Arrays.asList(new TaskJson("dummy"));
  }

}

File: haru-engine/src/main/kotlin/haru/engine/port/adapter/web/TaskController.kt
package haru.engine.port.adapter.web

import haru.engine.api.task.Task
import haru.engine.api.task.TaskService
import org.springframework.web.bind.annotation.RestController

/**
 * @author takeshi
 */
@RestController
open class TaskController : TaskService {

  override fun getTasks(): List<Task> {
    return listOf(Task("dummy"))
  }

}
